src/main/kotlin/io/arrowkt/arrowio/app/app.kt

```diff
@@ -48,17 +46,15 @@ fun usecase1() {
         a
     }.fix()
 
-    val y = c.run(AccountInMemoryRepository())
+    val y = c.run(AccountInMemoryRepository()).fix()
 
-    y.value().fix().unsafeRunAsync { cb ->
+
+
+    y.unsafeRunAsyncEither { cb ->
         cb.fold(
             { it.printStackTrace() },
-            { either ->
-                when (either) {
-                    is Either.Left -> println(either.a)
-                    is Either.Right -> either.b.forEach(::println)
-                }
-            }
+            { ex -> println(ex) },
+            { it.forEach(::println) }
```

Comment:
Nit: this could be an effect instead :D

Reply:
I think the whole main method is impure, otherwise there would be a single IO that is run. I would keep it like this.